front/src/pages/people/People.tsx

```diff
@@ -21,6 +22,16 @@ const StyledTableContainer = styled.div`
 export function People() {
   const [insertOnePerson] = useInsertOnePersonMutation();
 
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  useEffect(() => {
+    // Cancel the timer if the component unmounts before the timer expires
+    return () => {
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      if (timerRef.current) clearTimeout(timerRef.current);
```

Comment:
I move the timer up to the parent to prevent it from being accidentally cleared when the children re-render. We only want the timer to be cleared in case the user changes the page, this will happen when the parent at the top level unmounts.